### PR TITLE
WL-3566 : Add email sent confirmation message to feedback tool

### PR DESCRIPTION
**Pull request by nicholaswilson100**
This is only for logged in users. WL-3576 is for non logged in users.

I'm grabbing the siteUpdaters after form submission in the js and showing it in an info box at the top.

For the technical toAddress, same thing except that the email address is normally got in Java after the submission so I'm using the same one-liner to put it in the request before, before getting it out again in the js afterwards.

I'm re-using as much of the css for the error message as possible.

### CODE REVIEW

File: src/webapp/js/feedback.js [hunk 1]
```diff
@@ -244,6 +256,20 @@
         }
     };
 
+
+    feedback.displayInfo = function (siteUpdater) {
+		if (siteUpdater!=null && siteUpdater!=''){
```
Comment (buckett) [inline src/webapp/js/feedback.js]: Indentation

**Comment by nicholaswilson100** [comment]
Thanks.

File: src/webapp/js/feedback.js [hunk 2]
```diff
@@ -244,6 +256,20 @@
         }
     };
 
+
+    feedback.displayInfo = function (siteUpdater) {
+        if (siteUpdater!=null && siteUpdater!=''){
+            $('#feedback-info-message-wrapper span').html('An email with the information you entered has been sent to ' + siteUpdater);
+
+            $('#feedback-info-message-wrapper a').click(function (e) {
```
Comment (buckett) [inline src/webapp/js/feedback.js]: This is static setup rather than something that should be run each time.

File: src/webapp/js/feedback.js [hunk 3]
```diff
@@ -75,6 +84,9 @@
                 });
 
                 feedback.fitFrame();
+
```
Comment (buckett) [inline src/webapp/js/feedback.js:86]: Don't you want todo the fitFrame after showing the new html.

File: src/webapp/js/feedback.js [hunk 4]
```diff
@@ -75,6 +84,9 @@
...
+                feedback.displayInfo(siteUpdater);
+                feedback.displayInfo(technicalToAddress);
```
Comment (buckett) [inline src/webapp/js/feedback.js]: Doesn't this mean is only ever shows the technicalToAddress (as long as it's set in the server config)?

**Comment by buckett** [comment]
Not looked to closely, but I'd be tempted to have the server component return the email address used and then use this so the logic isn't duplicated between server and client about what address was used.

File: src/webapp/js/feedback.js (hunk 3 above)
Comment (nicholaswilson100) [inline src/webapp/js/feedback.js:86]: Yes you#re right

**Comment by nicholaswilson100** [comment]
I was keen not to duplicate any logic returning the email address but since it's only a one-liner 
(sakaiProxy.getConfigString(Constants.PROP_TECHNICAL_ADDRESS, null)) I thought not to bother. Don't mind factoring it out though if you think it's worth it...

File: src/webapp/js/feedback.js (hunk 2 above)
Comment (nicholaswilson100) [inline src/webapp/js/feedback.js]: Yup

File: src/webapp/js/feedback.js (hunk 4 above)
Comment (nicholaswilson100) [inline src/webapp/js/feedback.js]: Yes should really be if-else (I've fixed below)

File: src/webapp/js/feedback.js [hunk 5]
```diff
@@ -74,6 +83,18 @@
                                             } });
                 });
 
+
+                $('#feedback-info-message-wrapper a').click(function (e) {
+                    $('#feedback-info-message-wrapper').hide();
+                });
+
+                if (siteUpdater!=null && siteUpdater!='') {
```
Comment (buckett) [inline src/webapp/js/feedback.js:91]: Doesn't this happen when the page is ready which leads to it displaying a message that the email has been sent?

Comment (nicholaswilson100) [inline src/webapp/js/feedback.js:91]: Yup this happens when the page is ready but the message only shows if the user has either selected a site maintainer or if there is a hidden technicalToAddress field on the page, so it doesn't display the message straight away.

Comment (buckett) [inline src/webapp/js/feedback.js:91]: Ok so the first time the page renders (the HOME state) there isn't a selected maintainer or a technical contact element in the page, so despite calling displayInfo() there is never anything to display so it just leaves the page alone. 

Then when the technical content form has been submitted it still has the technical contact element in the page and so the switchstate will grab out the email address before it replaces the HTML content with the home state template.

**Pull request merged**